spp_base/models/reg_id.py

```diff
@@ -0,0 +1,20 @@
+# Part of OpenSPP. See LICENSE file for full copyright and licensing details.
+import logging
+import textwrap
+
+from odoo import _, api, fields, models
+from odoo.exceptions import ValidationError
+
+_logger = logging.getLogger(__name__)
+
+
+class OpenSPPRegistrantID(models.Model):
+    _inherit = "g2p.reg.id"
+
+    card_uid = fields.Char("Card UID")
+
+    @api.constrains("card_uid")
```

Comment:
it should be a constrains for both `id_type` & `card_uid` since constrains will be triggered after changing events come to DB --> you can create an ID with more or less 10 chars/digits UID number then change the id_type, nothing will stop you with this constrains only.

Reply:
Done